ansible/playbooks/restore-backup.yml

```diff
@@ -0,0 +1,106 @@
+---
+# This playbook can be run on staging to test the integrity of the backup, and
+# in production to set up a new production environment. Because it always needs
+# access to the production AWS bucket, when running on staging you need to add
+# "--vault-id production@prompt" to the run-playbook.sh call!
+- hosts: "all"
+  user: "ansible"
+  become: true
+  become_user: "root"
+  become_method: "sudo"
+  force_handlers: true
+
+  vars_prompt:
+    - name: "confirm"
+      default: 'ABORT'
+      private: false
+      prompt: |-
+        This playbook will restore a backup of the production server. This will
+        replace the data in many directories. There should be twice the size of
+        this backup available as free disk space, since the archives have to be
+        extracted.
+        Assume you will lose *any existing data* on this server. Are you
+        CERTAIN that you wish to continue? If so, enter "obliteration". Any
+        other value will abort
+
+  tasks:
+    - name: "check that confirmation is given"
+      assert:
+        that:
+          - "confirm == 'obliteration'"
+
+    # We only need this when running on staging, to access
+    # the AWS bucket later that has the production backups
+    - name: "load secrets from production"
+      include_vars:
+        name: "production"
+        file: "../group_vars/production/vault.yml"
+      when: "'production' not in group_names"
+
+    - name: "create temporary directory to store backup archives"
+      tempfile:
+          prefix: "backup-restore-test."
```

Comment:
This should:

 - Probably not contain `test`
 - Maybe include a timestamp?

Reply:
I'll clarify the name (`prod-backup-restore`), but a timestamp won't really have an added value since the module already adds a random suffix, and the timestamp is also in the metadata. And the entire dir should be deleted at the end of the run :P